patch/backend/app/engine/loaders/file.py

```diff
@@ -0,0 +1,61 @@
+import os
+import logging
+from llama_parse import LlamaParse
+from pydantic import BaseModel, validator
+
+logger = logging.getLogger(__name__)
+
+
+class FileLoaderConfig(BaseModel):
+    data_dir: str = "data"
+    use_llama_parse: bool = False
+
+    @validator("data_dir")
+    def data_dir_must_exist(cls, v):
+        if not os.path.isdir(v):
+            raise ValueError(f"Directory '{v}' does not exist")
+        return v
+
+
+def llama_parse_parser():
+    if os.getenv("LLAMA_CLOUD_API_KEY") is None:
+        raise ValueError(
+            "LLAMA_CLOUD_API_KEY environment variable is not set. "
+            "Please set it in .env file or in your shell environment then run again!"
+        )
+    parser = LlamaParse(result_type="markdown", verbose=True, language="en")
+    return parser
+
+
+def get_file_documents(config: FileLoaderConfig):
+    from llama_index.core.readers import SimpleDirectoryReader
+
+    import nest_asyncio
+
+    nest_asyncio.apply()
```

Comment:
Apply nest_asyncio. See: https://github.com/run-llama/llama_parse?tab=readme-ov-file#getting-started

Reply:
ideally we do this without patch - if we need a patch, better release a new upstream first

Reply:
It's now using create-llama 0.1.10 so i removed the patch.